python/turbine_models/tests/vmfb_comparison.py

```diff
@@ -0,0 +1,183 @@
+import os
+import sys
+import re
+
+from typing import Tuple
+
+os.environ["TORCH_LOGS"] = "dynamic"
+from transformers import AutoTokenizer, AutoModelForCausalLM
+import torch
+from torch.utils import _pytree as pytree
+from shark_turbine.aot import *
+from iree.compiler.ir import Context
+from iree import runtime as ireert
+
+from turbine_models.custom_models import remap_gguf
+import safetensors
+
+from tqdm import tqdm
+
+BATCH_SIZE = 1
+MAX_STEP_SEQ = 4095
+
+
+def torch_token_generator(
+    prompt, hf_model_name: str, hf_auth_token: str, break_on_eos=False
+):
+    tokenizer = AutoTokenizer.from_pretrained(
+        hf_model_name, use_fast=False, use_auth_token=hf_auth_token
+    )
+    model = AutoModelForCausalLM.from_pretrained(
+        hf_model_name, torch_dtype=torch.float, use_auth_token=hf_auth_token
+    )
+
+    initial_input = tokenizer(prompt, return_tensors="pt")
+    input_ids = initial_input.input_ids
+    past_key_values = None
+
+    while True:
+        model_results = model.forward(input_ids, past_key_values=past_key_values)
+        logits = model_results.logits
+        next_token_id = torch.argmax(logits[:, -1, :], dim=1)
+        past_key_values = model_results.past_key_values
+
+        yield next_token_id
+        input_ids = next_token_id.unsqueeze(0)  # Prepare for the next iteration
+
+        if next_token_id.item() == tokenizer.eos_token_id and break_on_eos:
```

Comment:
This will only provide a single sentence output right? The output we generate is multiple sentences. Also if break_on_eos=False will this get stuck in the while loop?